File: Tasks/Domain/Steps/Step.cs
namespace Domain.Steps;

public class Step
{
    public Guid Id { get; }
    public String Name { get; }
    public Boolean IsLast { get; }
    public Int32 Position { get; }

    public Step(Guid id, String name, Boolean isLast, Int32 position)
    {
        Id = id;
        Name = name;
        IsLast = isLast;
        Position = position;
    }
}

File: ClientApp/src/Domain/Steps/Models/Step.ts
export class Step {
    constructor(
        public readonly id: string,
        public readonly name: string,
        public readonly isLast: boolean,
        public readonly position: number
    ) {

    }
}

export function mapToStep(data: any): Step {
    return new Step(data.id, data.name, data.isLast, data.position)
}

export function mapToSteps(data: any[]): Step[] {
    return data.map(data => mapToStep(data))
}